src/Microsoft.Sbom.Api/Output/Telemetry/TelemetryRecorder.cs

```diff
@@ -46,6 +47,66 @@ public TelemetryRecorder(IFileSystemUtils fileSystemUtils, IConfiguration config
             Log = log ?? throw new ArgumentNullException(nameof(log));
         }
 
+        private TelemetryRecorder(string telemetryFilePath, IFileSystemUtils fileSystemUtils)
+        {
+            this.telemetryFilePath = telemetryFilePath;
+            FileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        }
+
+        public static TelemetryRecorder Create(string telemetryFilePath, IFileSystemUtils fileSystemUtils)
+        {
+            return new TelemetryRecorder(telemetryFilePath, fileSystemUtils);
+        }
+
+        //Method to log telemetry in conditions when the tool is not able to start execution of workflow.
+        public async Task ConsoleLogger(Exception e)
+        {
+            var logger = new LoggerConfiguration().WriteTo.Console().CreateLogger();
+            try
+            {
+                //convert exception to IDictionary<string,string> and handle unicode char.
+                var exceptions = new Dictionary<string, string>
+                {
+                    { e.GetType().ToString().Replace("\u0027", "'"), e.Message }
```

Comment:
Converting the exception was to match the format of the SBOMTelemetry object. and replacing the unicode character was so the quotes would display properly around a given file path. We also have the option to create a new property inside the SBOMTelemetry object to store an entire Exception and use this instead. It would look like this 

![telemetrynew](https://user-images.githubusercontent.com/69322674/228966537-e93e11d2-3647-4084-97da-97455034c121.png)

Reply:
We can do either one although I'm not sure how useful all the other extra information would be in this scenario.

Reply:
But this is just the Type of the Exception, right? Does the type contain quote characters?